base/abstractarray.jl

```diff
@@ -644,8 +662,9 @@ _getindex{T,N}(::LinearSlow, A::AbstractArray{T,N}, I::Vararg{Real, N}) = (@_pro
 function _getindex(::LinearSlow, A::AbstractArray, i::Real)
     # ind2sub requires all dimensions to be > 0; may as well just check bounds
     @_inline_meta
-    @boundscheck checkbounds(A, i)
```

Comment:
Why not keep this as `checkbounds`? Won't this make custom behaviors harder to implement?